stock_move_location/wizard/stock_move_location.py

```diff
@@ -0,0 +1,390 @@
+            ("company_id", "=", False),
+        ]
+
+    def _create_picking(self):
+        return self.env["stock.picking"].create(
+            {
+                "picking_type_id": self.picking_type_id.id,
+                "location_id": self.origin_location_id.id,
+                "location_dest_id": self.destination_location_id.id,
+            }
+        )
+
+    def group_lines(self):
+        lines_grouped = {}
+        for line in self.stock_move_location_line_ids:
+            lines_grouped.setdefault(
+                line.product_id.id, self.env["wiz.stock.move.location.line"].browse()
+            )
+            lines_grouped[line.product_id.id] |= line
+        return lines_grouped
+
+    def _create_moves(self, picking):
+        self.ensure_one()
+        groups = self.group_lines()
+        moves = self.env["stock.move"]
+        for lines in groups.values():
+            moves |= self._create_move(picking, lines)
+        return moves
+
+    def _get_move_values(self, picking, lines):
+        # locations are same for the products
+        location_from_id = lines[0].origin_location_id.id
+        location_to_id = lines[0].destination_location_id.id
+        product = lines[0].product_id
+        product_uom_id = lines[0].product_uom_id.id
+        qty = sum(x.move_quantity for x in lines)
+        return {
+            "name": product.display_name,
+            "location_id": location_from_id,
+            "location_dest_id": location_to_id,
+            "product_id": product.id,
+            "product_uom": product_uom_id,
+            "product_uom_qty": qty,
+            "picking_id": picking.id,
+            "location_move": True,
+        }
+
+    def _create_move(self, picking, lines):
+        self.ensure_one()
+        move = self.env["stock.move"].create(self._get_move_values(picking, lines))
+        lines.create_move_lines(picking, move)
+        if self.env.context.get("planned"):
+            for line in lines:
+                quants = self.env["stock.quant"]._gather(
+                    line.product_id,
+                    line.origin_location_id,
+                    lot_id=line.lot_id,
+                    package_id=line.package_id,
+                    owner_id=line.owner_id,
+                    strict=False,
+                    qty=line.move_quantity,
+                )
+                move._update_reserved_quantity(
+                    line.move_quantity,
+                    line.origin_location_id,
+                    quant_ids=quants,
+                    lot_id=line.lot_id,
+                    package_id=line.package_id,
+                    owner_id=line.owner_id,
+                    strict=False,
+                )
+            # Force the state to be assigned, instead of _action_assign,
+            # to avoid discarding the selected move_location_line.
+            move.state = "assigned"
+            move.move_line_ids.write({"state": "assigned"})
```

Comment:
Also it would be nice to remove empty lines (having quantity == 0) in case of planned transfers, something like this:

```suggestion
            move.state = "assigned"
            move.move_line_ids.filtered(lambda ml: not ml.quantity).unlink()
            move.move_line_ids.write({"state": "assigned"})
```